<h1>Activity #1 - Palmer Penguins Data Analytics</h1>
<hr>
<h3>Analyst: Jeremy Bryan T. Villanueva</h3>

In [1]:
#import Libraries
import numpy as np
import pandas as pd

<h3>Converting Dataset(s) into DataFrame</h3>
<h3>Requirements:</h3>
<ul>
    <li>Name of the DataFrame</li>
    <li>File Path</li>
</ul>

In [2]:
#import csv file
penguins = pd.read_csv('datasets\\penguins.csv')
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [3]:
# penguins.describe()
#.T for transpose or to flip
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Checking for Null Values</h3>

In [4]:
#pengruins.isnull()
#sum() for summarize
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Filling up the Missing Values</h3>

In [5]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [6]:
penguins['bill_length_mm'].mean()
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [8]:
penguins['bill_depth_mm'].mean()
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [9]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [10]:
penguins['flipper_length_mm'].mean()
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [11]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [12]:
penguins['body_mass_g'].mean()
penguins['body_mass_g'] = penguins['flipper_length_mm'].fillna(penguins['body_mass_g'].mean())

In [13]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [14]:
penguins['sex'].mode()
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [15]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h3>Changing the Data Type</h3>

In [16]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [17]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')

<h3>Making a column</h3>

In [18]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [74]:
#it assumes you will make a new column
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 1)
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0       1     Adelie  Torgersen        39.10000       18.70000   
1       2     Adelie  Torgersen        39.50000       17.40000   
2       3     Adelie  Torgersen        40.30000       18.00000   
3       4     Adelie  Torgersen        43.92193       17.15117   
4       5     Adelie  Torgersen        36.70000       19.30000   
..    ...        ...        ...             ...            ...   
339   340  Chinstrap      Dream        55.80000       19.80000   
340   341  Chinstrap      Dream        43.50000       18.10000   
341   342  Chinstrap      Dream        49.60000       18.20000   
342   343  Chinstrap      Dream        50.80000       19.00000   
343   344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
0           181.000000  3750.000000    male  2007      3.8  
1           186.000000  3800.000000  female  2007      3.8  
2           195.000000  3250.000000  female  2007      3.2  
3           200.915205  4201.754386    male  2007      4.2  
4           193.000000  3450.000000  female  2007      3.4  
..                 ...          ...     ...   ...      ...  
339         207.000000  4000.000000    male  2009      4.0  
340         202.000000  3400.000000  female  2009      3.4  
341         193.000000  3775.000000    male  2009      3.8  
342         210.000000  4100.000000    male  2009      4.1  
343         198.000000  3775.000000  female  2009      3.8  

[344 rows x 10 columns]

<h3>Creating individual DataFrames for each species.</h3>

In [20]:
species1 = penguins[penguins['species'] == 'Chinstrap']
species2 = penguins[penguins['species'] == 'Adelie']
species3 = penguins[penguins['species'] == 'Gentoo']
# species1
# species2
# species3

<h2>Explanatory Data Analytics</h2>
<hr>
<h3>Question #1</h3>
<h4>Which species has the greatest bill depth?</h4>

In [41]:
penguins['bill_depth_mm'].max()

np.float64(21.5)

In [42]:
penguins[penguins['bill_depth_mm'] == 21.5]

rowid species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
19    20  Adelie  Torgersen            46.0           21.5              194.0   

    body_mass_g   sex  year  
19       4200.0  male  2007

<h3>Insight:</h3>
<p>The species that has the longest bill is the Adélie penguin.</p>
<hr>

<h3>Question #2</h3>
<h4>In which year were the fewest penguin records observed?</h4>

In [43]:
penguins["year"].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h3>Insight:</h3>
<p>The year 2007 has the fewest penguin records.</p>
<hr>

<h3>Question #3</h3>
<h4>Which gender of penguins is most common on Dream Island?</h4>

In [67]:
penguins[(penguins['island'] == 'Dream') & (penguins['sex'] == 'male')].value_counts()

rowid  species    island  bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  sex   year
32     Adelie     Dream   37.2            18.1           178.0              3900.0       male  2007    1
34     Adelie     Dream   40.9            18.9           184.0              3900.0       male  2007    1
36     Adelie     Dream   39.2            21.1           196.0              4150.0       male  2007    1
37     Adelie     Dream   38.8            20.0           190.0              3950.0       male  2007    1
40     Adelie     Dream   39.8            19.1           184.0              4650.0       male  2007    1
                                                                                                      ..
335    Chinstrap  Dream   50.2            18.8           202.0              3800.0       male  2009    1
337    Chinstrap  Dream   51.9            19.5           206.0              3950.0       male  2009    1
340    Chinstrap  Dream   55.8            19.8           207

In [68]:
penguins[(penguins['island'] == 'Dream') & (penguins['sex'] == 'female')].value_counts()

rowid  species    island  bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  sex     year
31     Adelie     Dream   39.5            16.7           178.0              3250.0       female  2007    1
33     Adelie     Dream   39.5            17.8           188.0              3300.0       female  2007    1
35     Adelie     Dream   36.4            17.0           195.0              3325.0       female  2007    1
38     Adelie     Dream   42.2            18.5           180.0              3550.0       female  2007    1
39     Adelie     Dream   37.6            19.3           181.0              3300.0       female  2007    1
                                                                                                        ..
336    Chinstrap  Dream   45.6            19.4           194.0              3525.0       female  2009    1
338    Chinstrap  Dream   46.8            16.5           189.0              3650.0       female  2009    1
339    Chinstrap  Dream   45.7            

<h3>Insight:</h3>
<p>Male is the most common gender among penguins on Dream Island.</p>
<hr>

<h3>Question #4</h3>
<h4>On which island is the heaviest penguin found?</h4>

In [52]:
penguins[penguins["body_mass_g"] == penguins["body_mass_g"].max()]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169   170  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex  year  
169       6300.0  male  2007

<h3>Insight:</h3>
<p>In 2007, a male Gentoo penguin was recorded on Biscoe Island with a bill length of 49.2 mm, a bill depth of 15.2 mm, a flipper length of 221 mm, and a body mass of 6,300 grams.</p>
<hr>

<h3>Question #5</h3>
<h4>In which year was the shortest flipper length in a penguin recorded?</h4>

In [62]:
penguins['flipper_length_mm'].min()

np.float64(172.0)

In [63]:
penguins[penguins['flipper_length_mm'] == 172.0]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
28    29  Adelie  Biscoe            37.9           18.6              172.0   

    body_mass_g     sex  year  
28       3150.0  female  2007

<h3>Insight:</h3>
<p>The shortest flipper length penguin was recorded in 2007; it was a female Adélie penguin.</p>
<hr>

<h3>Question #6</h3>
<h4>How many penguins are there in each species?</h4>

In [73]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>Insight:</h3>
<p>There are 152 Adelie, 124 Gentoo, and 68 Chinstrap penguins recorded</p>
<hr>

<h3>Question #7</h3>
<h4>Which penguin species is the lightest in kilograms?</h4>

In [77]:
penguins['mass_kg'].min()

np.float64(2.7)

In [78]:
penguins[penguins['mass_kg'] == 2.7]

rowid    species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
314   315  Chinstrap  Dream            46.9           16.6              192.0   

     body_mass_g     sex  year  mass_kg  
314       2700.0  female  2008      2.7

<h3>Insight:</h3>
<p>In 2008, a female Chinstrap penguin on Dream Island was recorded with a body mass of only 2.7 kg, making it one of the lightest penguins recorded.</p>
<hr>